software/communication.py

```diff
@@ -0,0 +1,35 @@
+import serial
+import struct
+
+#ports=serial.tools.list_ports.comports()
+#print([port.name for port in ports])
+
+class Communication:
+    def __init__(self):
+        self.ser = serial.Serial()
+        self.ser.port='/dev/ttyACM0'

```

Comment:
Still using a hardcoded serial port. Refer to:
https://ut-robotics.github.io/picr22-home/basketball_robot_guide/software/review_summary.html#_use_serial_tools_list_ports_or_equivalent_to_find_correct_serial_port

Reply:
Now it is not hardcoded and i think it works.